apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -49,9 +49,11 @@ export class SendMessageInApp extends SendMessageType {
       );
     }
 
-    if (inAppChannel.template.cta?.actions) {
-      inAppChannel.template.cta?.actions.map((action) => {
-        if (action.content.text) return contentService.replaceVariables(action.content.text, messageVariables);
+    if (inAppChannel.template.cta?.action?.buttons) {
+      inAppChannel.template.cta.action.buttons = inAppChannel.template.cta.action.buttons.map((action) => {
+        if (!action.content) return action;
+
+        return { type: action.type, content: contentService.replaceVariables(action.content, messageVariables) };
```

Comment:
Check out the compileTemplate usecase for this, so we can support nested properties.

Reply:
I do not see any use of the 'compileTemplate' other than the email usecase, if there is a  place I can see the new support of the nested templateCompile objects that you were talking about @davidsoderberg  in the discord server?